### PR TITLE
Refactor auth routing. Add new Jenkins ARM template

### PR DESCRIPTION
**Pull request by fashaikh**

### CODE REVIEW

File: SimpleWAWS/Authentication/SecurityManager.cs [hunk 1]
```diff
@@ -221,5 +222,29 @@ private static bool ValidDateTimeSessionCookie(DateTime date)
...
+                var state = context.Request["state"];
+                var redirectlocation = state.Split('?')[0];
+                return $"{redirectlocation}?cookie={cookie}&state={Uri.EscapeDataString(state)}";
+            }
+            else
+            {
+                return context.Request["state"];
+            }
+        }
+        public static string CreateTrySessionCookie(IPrincipal user)
```
Comment (ahmelsayed) [inline SimpleWAWS/Authentication/SecurityManager.cs]: refactor the code so that we don't have 2 places where we create cookies

Comment (fashaikh) [inline SimpleWAWS/Authentication/SecurityManager.cs]: Refactored.. Look good ?

File: SimpleWAWS/Authentication/SecurityManager.cs [hunk 2]
```diff
@@ -13,6 +13,7 @@
 using SimpleWAWS.Models;
 using System.Globalization;
 using System.Linq;
+using System.Security.Principal;
```
Comment (ahmelsayed) [inline SimpleWAWS/Authentication/SecurityManager.cs]: remove?

File: SimpleWAWS/Authentication/SecurityManager.cs (hunk 1 above)
Comment (ahmelsayed) [inline SimpleWAWS/Authentication/SecurityManager.cs]: yep, looks good

**Pull request merged**